### PR TITLE
Feature#794

### PR DESCRIPTION
**Pull request by magdyheiba**
Screenshot 2022-12-14 at 1 12 06 AM
Screenshot 2022-12-14 at 1 11 55 AM
Screenshot 2022-12-14 at 1 11 13 AM

### CODE REVIEW

File: src/main/resources/defaultProperties/WebCapabilities.properties [hunk 1]
```diff
@@ -10,6 +10,7 @@ mobileEmulation.width=
 mobileEmulation.height=
 mobileEmulation.pixelRatio=
 mobileEmulation.userAgent=
+BaseURL=https://demoqa.com/
```
Comment (MohabMohie) [inline src/main/resources/defaultProperties/WebCapabilities.properties]: can we convert to lower camel case please?
we also need to expose this property in the configuration manager :)

**Review by MohabMohie — CHANGES_REQUESTED** [review]
Thanks a lot for the great added value!
Let's consider using lower camel case and add to the configuration manager.
Do you want my support with that?

File: src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java [hunk 2]
```diff
@@ -230,6 +230,15 @@ private static String getDomainNameFromURL(String url) {
...
      */
     public static void navigateToURL(WebDriver driver, String targetUrl, String targetUrlAfterRedirection) {
+        // check if the user wrote the URL ended with "/"
+        String finalUrl = null;
+        if (System.getProperty("baseURL").endsWith("/")) {
+            finalUrl = System.getProperty("baseURL").substring(0, System.getProperty("baseURL").length() - 1);
+        }
+        // check if the user sends the URL with the abbreviation of "./"
+        if (targetUrl.startsWith(".")) {
+            targetUrl = targetUrl.replaceFirst(".", finalUrl);
```
Comment (MohabMohie) [inline src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java]: should this be "finalUrl =" ??

File: src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java [hunk 3]
```diff
@@ -830,41 +839,41 @@ public WebDriverBrowserActions fullScreenWindow() {
...
     }
-    
+
     /**
      * Switches focus to another Tab
      *
      * @param driver       the current instance of Selenium webdriver
      * @param URL The name of the URL you want to navigate to
      */
     public static void switchToNewTab(WebDriver driver, String URL) {
```
Comment (MohabMohie) [inline src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java]: formatting issue?

File: src/main/resources/defaultProperties/WebCapabilities.properties [hunk 4]
```diff
@@ -10,6 +10,7 @@ mobileEmulation.width=
 mobileEmulation.height=
 mobileEmulation.pixelRatio=
 mobileEmulation.userAgent=
+baseURL=https://demoqa.com
```
Comment (MohabMohie) [inline src/main/resources/defaultProperties/WebCapabilities.properties]: can we set the default to empty please? we also need to add some validations to ensure that "empty" doesn't throw any exceptions.

File: src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java (hunk 2 above)
Comment (magdyheiba) [reply src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java]: I changed it to Modified URL and added cases to handle If user selects to navigate with the new feature but didn't add a base Url

File: src/main/resources/defaultProperties/WebCapabilities.properties (hunk 4 above)
Comment (magdyheiba) [reply src/main/resources/defaultProperties/WebCapabilities.properties]: Screenshot 2022-12-14 at 3 36 11 PM

When a user selects to navigate with the new feature but didn't add a base Url , SHAFT will throw an error as described at the screenshot

File: src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java (hunk 3 above)
Comment (magdyheiba) [reply src/main/java/com/shaft/gui/browser/WebDriverBrowserActions.java]: Fixed el7amdoulelah through eclipse :)

File: src/main/resources/defaultProperties/WebCapabilities.properties (hunk 1 above)
Comment (magdyheiba) [reply src/main/resources/defaultProperties/WebCapabilities.properties]: Done

**Review by MohabMohie — APPROVED** [review]
Thanks a lot for your valuable contribution 🙏

**Pull request merged**